src/components/header/movie-search-header.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useEffect, useState } from 'react';
+import { connect } from 'react-redux';
+import { RouteProps } from 'react-router';
+import { withRouter } from 'react-router-dom';
+import styled from 'styled-components';
+
+import { createMovie, searchMovie } from '../../redux/actions/movie';
+import Logo from '../logo/logo';
+import Modal from '../modal/modal';
+import ModalAddEditContent from '../modal/modal-add-edit-content';
+import { Movie } from '../movies/movies';
+import { HeaderAddBtn } from './header-add-btn';
+import { HeaderSearchBlock } from './header-search-block';
+import { HeaderSearchBtn } from './header-search-btn';
+import { HeaderSearchInput } from './header-search-input';
+import { HeaderSearchLine } from './header-search-line';
+import { useStyles } from './header-styles';
+import { HeaderTitle } from './header-title';
+import { HeaderTop } from './header-top';
+
+interface Props {
+  className?: string;
+  createMovie?: (e: any) => void;
+  searchMovie?: (e: any) => void;
+  genresOptions?: string[];
+  history: RouteProps["history"];
+  location: RouteProps["location"];
+}
+
+const MovieSearchHeaderComponent = ({
+  className,
+  createMovie,
+  searchMovie,
+  genresOptions,
+  history,
+  location,
+}: Props) => {
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [inputValue, setInputValue] = useState(
+    location?.search?.split("=").slice(1).join()
+  );
+
+  // ? render
+  useEffect(() => {
```

Comment:
You don't need it here. Initial movies list loading is already done in movies container.
Get `searchQuery` with `useParams` and pass it to the `fetchMovies`